### ISSUE TITLE
GMap in Notebook unreliable again

### ISSUE THREAD

**Issue by bryevdv** [bokeh/bokeh#6328]
I believe (but have not yet confirmed) that this happened in the `0.12.6dev2` to `dev3` time frame, which would possibly implicate the big layout PR. 

Behavior is for plot areas to render solid grey on the first `show` but often render correctly after a save and reload of the notebook. No out of the ordingary JS console messages or errors. 

ping @mattpap

**Comment by jtaylor306** [comment]
I am having this issue:

![broken2](https://cloud.githubusercontent.com/assets/5560742/26472102/9d4bbf74-4172-11e7-979c-d9d411e0ded1.JPG)

The map load fine on reload, and they also load fine when I let it pop up into a new window. 
Python 2.7
IPython 4.1.2
Jupyter Notebook server 4.1.0
Windows 7
bokeh 0.12.6dev2 probably important to mention im having issues with other versions as well. 

Here is the example notebook that is not working(I've also tried other examples that some people confirmed work for them):
https://github.com/jtaylor306/JavaHelloWorld/blob/master/gmapbroken.ipynb

I'm going to keep trying some things and update this as a I figure out whats going on. I'll also give it a try on a different PC.

**Comment by bryevdv** [comment]
Here are log traces from a working reload (left) and a non-working first time `show`. Possibly notable difference, in the working case `[bokeh] – "initial ranges set"` happens *before* the call to `_build_map`


<img width="1550" alt="screen shot 2017-05-25 at 18 12 00" src="https://cloud.githubusercontent.com/assets/1078448/26474286/b617e214-4175-11e7-9981-2f914add000a.png">

**Comment by syonekura** [comment]
Similar behavior in Bokeh 0.12.5 (solid gray map), but if you resize the window containing the map (i did it accidentally when inspecting the noteboook by resizing the browser) the map reappears

**Comment by bryevdv** [comment]
I am unable to reproduce any problems on 0.12.5 so more details would be needed for that.

**Comment by jtaylor306** [comment]
I can confirm my issues with 0.12.5 are the same as the user syonekura reported. This is on an entire different machine as well. Windows 10 and Python 3 in this case.

**Comment by jtaylor306** [comment]
Can you inspect the map element in the browser in the working version? Does this error appear there too? I'm using chrome. 
![uncaught](https://cloud.githubusercontent.com/assets/5560742/26512670/92129c90-4235-11e7-999e-f7f733a60deb.JPG)

**Comment by syonekura** [comment]
I'm sorry, I said similar in the sense that I get the same output (gray map), but with a different workflow on 0.12.5, I should've specified that. So maybe this is another bug, but I guess that's related to this one (and hope that it helps in someway to detect it).

In 0.12.5 the first run plots the map without problems. The second run shows a gray map. This is the code I used:

```
from __future__ import print_function

from bokeh.io import show, output_notebook
from bokeh.document import Document
from bokeh.embed import file_html
from bokeh.models.glyphs import Circle
from bokeh.models import (
    GMapPlot, Range1d, ColumnDataSource, PanTool, WheelZoomTool, BoxSelectTool, GMapOptions,
    LinearAxis, MercatorTickFormatter, MercatorTicker, Label)
from bokeh.resources import INLINE

output_notebook()

x_range = Range1d()
y_range = Range1d()

# JSON style string taken from: https://snazzymaps.com/style/1/pale-dawn
map_options = GMapOptions(lat=30.2861, lng=-97.7394, map_type="roadmap", zoom=13, styles="""
[{"featureType":"administrative","elementType":"all","stylers":[{"visibility":"on"},{"lightness":33}]},{"featureType":"landscape","elementType":"all","stylers":[{"color":"#f2e5d4"}]},{"featureType":"poi.park","elementType":"geometry","stylers":[{"color":"#c5dac6"}]},{"featureType":"poi.park","elementType":"labels","stylers":[{"visibility":"on"},{"lightness":20}]},{"featureType":"road","elementType":"all","stylers":[{"lightness":20}]},{"featureType":"road.highway","elementType":"geometry","stylers":[{"color":"#c5c6c6"}]},{"featureType":"road.arterial","elementType":"geometry","stylers":[{"color":"#e4d7c6"}]},{"featureType":"road.local","elementType":"geometry","stylers":[{"color":"#fbfaf7"}]},{"featureType":"water","elementType":"all","stylers":[{"visibility":"on"},{"color":"#acbcc9"}]}]
""")

# Google Maps now requires an API key. You can find out how to get one here:
# https://developers.google.com/maps/documentation/javascript/get-api-key
API_KEY = 'API KEY'

plot = GMapPlot(
    x_range=x_range, y_range=y_range,
    map_options=map_options,
    api_key=API_KEY,
)

plot.title.text = "Austin"

source = ColumnDataSource(
    data=dict(
        lat=[30.2861, 30.2855, 30.2869],
        lon=[-97.7394, -97.7390, -97.7405],
        fill=['orange', 'blue', 'green']
    )
)

circle = Circle(x="lon", y="lat", size=15, fill_color="fill", line_color="black")
plot.add_glyph(source, circle)

pan = PanTool()
wheel_zoom = WheelZoomTool()
box_select = BoxSelectTool()

plot.add_tools(pan, wheel_zoom, box_select)

xformatter = MercatorTickFormatter(dimension="lon")
xticker = MercatorTicker(dimension="lon")
xaxis = LinearAxis(formatter=xformatter, ticker=xticker)
plot.add_layout(xaxis, 'below')

yformatter = MercatorTickFormatter(dimension="lat")
yticker = MercatorTicker(dimension="lat")
yaxis = LinearAxis(formatter=yformatter, ticker=yticker)
plot.add_layout(yaxis, 'left')

show(plot)
```

**Comment by bryevdv** [comment]
@jtaylor306 in your output it's definitely trying to render the map before the Google Maps API is loaded.

**Comment by mattpap** [comment]
I can't reproduce any of errors in this issue on `master`.

**Comment by bryevdv** [comment]
@mattpap maybe your last PR coincidentally fixed, I will check now. I was definitely having problems on master yesterday.

**Comment by bryevdv** [comment]
Still broken with inline resources for me:

<img width="703" alt="screen shot 2017-05-30 at 08 54 02" src="https://cloud.githubusercontent.com/assets/1078448/26586687/9a50e712-4515-11e7-9a20-490f93cd7279.png">

**Comment by mattpap** [comment]
@bryevdv, is this a notebook from our examples?

**Comment by bryevdv** [comment]
No but it as based on one. Here is the entire thing (three cells):
```
from bokeh.io import output_file, show, output_notebook
from bokeh.models import (
  GMapPlot, GMapOptions, ColumnDataSource, Circle, Range1d, PanTool, WheelZoomTool, BoxSelectTool
)
output_notebook()
```

```
map_options = GMapOptions(lat=30.5, lng=-84.73, map_type="roadmap", zoom=13)

plot = GMapPlot(
    x_range=Range1d(), y_range=Range1d(), map_options=map_options
)
plot.title.text = "Austin"

# For GMaps to function, Google requires you obtain and enable an API key:
#
#     https://developers.google.com/maps/documentation/javascript/get-api-key
#
# Replace the value below with your personal API key:
plot.api_key = "API KEY HERE"

source = ColumnDataSource(
    data=dict(
        lat=[30.51387, 30.19048, 30.16894, 30.50908, 30.50769, 30.49269, 30.43251, 30.573485, 30.63863, 30.44942, 30.51089, 30.170405, 30.181025, 30.4114, 30.57711, 30.649405, 30.51381, 30.543535, 30.40602, 30.17168, 30.463155, 30.34513, 30.64536, 30.46154, 30.41773, 30.59529, 30.41557, 30.507335, 30.60825, 30.190665, 30.46403, 30.40493, 30.196925, 30.52122, 30.430805, 30.29582, 30.61521, 30.581515, 30.518045, 30.61865, 30.55107, 30.10837, 30.529115, 30.044845, 30.166955, 30.5925, 30.507405, 30.06169, 30.44163, 30.44739, 30.505485, 30.06159, 30.62498, 30.14499, 30.192645, 30.60246, 30.32025, 30.422005, 30.237775, 30.59735, 30.48061, 30.3905, 30.61697, 30.42737, 30.60892],
        lon=[-84.7025, -84.37763, -84.301085, -84.12473, -84.47559, -84.22631, -84.524, -84.82331, -84.37893, -84.08317, -84.481805, -84.294305, -84.297735, -84.139825, -84.35903, -84.401355, -84.33922, -84.253045, -84.09648, -84.298555, -84.089005, -84.396665, -84.38444, -84.1127, -84.155385, -84.69077, -84.13896, -84.484485, -84.66413, -84.34237, -84.08307, -84.97252, -84.34312, -84.3429, -84.415115, -84.3141, -84.5212, -84.230575, -84.360625, -84.263265, -84.21563, -84.342705, -84.362745, -85.078905, -84.316875, -84.62054, -84.09067, -84.28916, -84.40821, -84.36664, -84.054955, -84.291435, -84.384285, -84.34291, -84.36448, -84.471315, -84.313395, -84.526495, -85.035555, -84.501125, -84.418455, -84.138615, -84.229765, -84.51276, -84.66374],
    )
)

circle = Circle(x="lon", y="lat", size=15, fill_color="blue", fill_alpha=0.8, line_color=None)
plot.add_glyph(source, circle)

plot.add_tools(PanTool(), WheelZoomTool(), BoxSelectTool())
```

```
show(plot)
```

**Comment by bryevdv** [comment]
@mattpap I'll take a look at this tomorrow since you can't reproduce. My concern is just clashing with any layout code that is not in a branch/PR yet.

**Comment by bryevdv** [comment]
I believe this is fixed in #6361 I am unable to reproduce the "grey box" problem in the notebook despite various ways of trying. There will be another dev build soon however it would be good to get more people testing things.

**Comment by tbrandon** [comment]
Bokeh 0.12.13
I'm getting a "grey box" map, when a map is placed in tab that is not active.  Switching to the tab with the map shows the "grey box", with Google's logo in the bottom left corner and "Terms of Use" in the bottom right corner.  With the "map tab" active, resizing the browser make the map appear correctly.  If the map is placed in the tab that is active on load the map appears correctly.  This behaviour is observed in Chrome and Firefox.  Thank you.  

```
from bokeh.models.layouts import Column
from bokeh.models import (
    GMapPlot, GMapOptions, Range1d 
)
from bokeh.models.widgets import Panel, Tabs
from bokeh.plotting import curdoc, figure

class GreyMap:
    def __init__(self):
        map_options = GMapOptions(lat=30.29, lng=-97.73, map_type="roadmap", zoom=11)

        plot = GMapPlot(x_range=Range1d(), y_range=Range1d(), map_options=map_options)

        # For GMaps to function, Google requires you obtain and enable an API key:
        #
        #     https://developers.google.com/maps/documentation/javascript/get-api-key
        #
        # Replace the value below with your personal API key:
        plot.api_key = "GOOGLE_API_KEY"

        panel_1 = Panel(title='Tab 1', child=Column())
        panel_2 = Panel(title='Map Tab', child=Column(plot))
        self.tabs = Tabs(width=1000, tabs=[panel_1, panel_2])

inst = GreyMap()
curdoc().add_root(inst.tabs)
``` 
bokeh serve greymap.py
